src/lamp_py/aws/s3.py

```diff
@@ -555,48 +553,47 @@ def get_datetime_from_partition_path(path: str) -> datetime.datetime:
     return return_date
 
 
-def _get_pyarrow_table(
+def _get_pyarrow_dataset(
     filename: Union[str, List[str]],
-    filters: Optional[Union[Sequence[Tuple], Sequence[List[Tuple]]]] = None,
-) -> pyarrow.Table:
+    filters: Optional[pd.Expression] = None,
+) -> pd.Dataset:
     """
-    internal function to get pyarrow table from parquet file(s)
+    internal function to get pyarrow dataset from parquet file(s)
     """
-
     active_fs = fs.S3FileSystem()
 
     if isinstance(filename, list):
         to_load = [f.replace("s3://", "") for f in filename]
     else:
         to_load = [filename.replace("s3://", "")]
 
-    # using `read_pandas` because `read` with "columns" parameter results in
-    # much slower file downloads for some reason...
-    return pq.ParquetDataset(
-        to_load, filesystem=active_fs, filters=filters
-    ).read_pandas()
+    ds = pd.dataset(to_load, filesystem=active_fs)
+    if filters is not None:
+        ds = ds.filter(filters)
+
+    return ds
```

Comment:
Previously, we were materializing these parquet files as pyarrow tables. This new flow first returns a dataset (lazy), with filters applied, and then materializes the pyarrow table or creates record batches depending on the use case.